hazelcast/src/test/java/com/hazelcast/test/TestTaskExecutorUtil.java

```diff
@@ -19,68 +19,57 @@
 
 /**
  * Convenience for executing {@link Callable} on a partition thread.
- * Unlike regular {@link PartitionSpecificRunnable} it's easy to return a value back to a caller.
- *
+ * <p>
+ * Unlike the regular {@link PartitionSpecificRunnable} it's easy to return a
+ * value back to a caller.
+ * <p>
  * This is intended to be used in tests only.
- *
  */
 public final class TestTaskExecutorUtil {
 
-    private static final Object NULL_VALUE = new Object();
     private static final int TIMEOUT_SECONDS = 120;
+    private static final Object NULL_VALUE = new Object();
 
     private TestTaskExecutorUtil() {
-
     }
 
     /**
-     * Executes a callable on a specific partition thread and return a result.
-     * This does NOT check if a given Hazelcast instance owns a specific partition.
+     * Executes a {@link Callable} on a specific partition thread and returns
+     * a result.
+     * <p>
+     * This does <b>not</b> check if a given Hazelcast instance owns a specific
+     * partition.
      *
-     * @param instance Hazelcast instance to be used for task executin
-     * @param task the task to be executed
+     * @param instance    Hazelcast instance to be used for task execution
+     * @param task        the task to be executed
      * @param partitionId selects partition thread
-     * @param <T> type of the result
+     * @param <T>         type of the result
      * @return result as returned by the callable
      */
     public static <T> T runOnPartitionThread(HazelcastInstance instance, final Callable<T> task, final int partitionId) {
         InternalOperationService operationService = getNodeEngineImpl(instance).getOperationService();
-        final BlockingQueue<Object> resultQueue = new ArrayBlockingQueue<Object>(1);
-        operationService.execute(new PartitionSpecificRunnable() {
-            @Override
-            public int getPartitionId() {
-                return partitionId;
-            }
-
-            @Override
-            public void run() {
-                try {
-                    Object result = wrapNullIfNeeded(task.call());
-                    resultQueue.add(result);
-                } catch (Throwable e) {
-                    resultQueue.add(e);
-                }
-            }
-        });
+        BlockingQueue<Object> resultQueue = new ArrayBlockingQueue<Object>(1);
+        operationService.execute(new PartitionSpecificRunnableWithResultQueue<T>(partitionId, task, resultQueue));
         try {
             Object result = resultQueue.poll(TIMEOUT_SECONDS, SECONDS);
             if (result instanceof Throwable) {
-                ExceptionUtil.sneakyThrow((Throwable) result);
+                sneakyThrow((Throwable) result);
             }
+            //noinspection unchecked
             return (T) unwrapNullIfNeeded(result);
         } catch (InterruptedException e) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
```

Comment:
This was the wrong method call to restore the interrupted flag.

Reply:
![image](https://user-images.githubusercontent.com/158619/42646273-428b1f20-85b5-11e8-877b-5d3ac854371e.png)

Reply:
This was (again) a hint from IntelliJ: a warning about unused return value from `Thread.interrupted()`